src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java

```diff
@@ -74,30 +75,38 @@ private boolean isReady(BlobItem blob, String containerName) {
     }
 
     private void processBlob(BlobClient blob) {
-        leaseAcquirer.ifAcquiredOrElse(
-            blob,
-            () -> {
-                envelopeService
-                    .findLastEnvelope(blob.getBlobName(), blob.getContainerName())
-                    .ifPresentOrElse(
+        Optional<Envelope> optionalEnvelope = envelopeService
+            .findLastEnvelope(blob.getBlobName(), blob.getContainerName());
+
+        if (canProcess(optionalEnvelope)) {
+            leaseAcquirer.ifAcquiredOrElse(
+                blob,
+                () -> {
+                    optionalEnvelope.ifPresentOrElse(
                         envelope -> handleBlobWithExistingEnvelope(blob, envelope),
                         () -> handleBlobWithoutEnvelope(blob)
                     );
-            },
-            () -> logger.info(
-                "Cannot acquire a lease for blob - skipping. File name: {}, container: {}",
-                blob.getBlobName(),
-                blob.getContainerName()
-            )
-        );
+                },
+                () -> logger.info(
+                    "Cannot acquire a lease for blob - skipping. File name: {}, container: {}",
+                    blob.getBlobName(),
+                    blob.getContainerName()
+                )
+            );
+        } else {
+            optionalEnvelope.ifPresent(
+                env -> logger.info("Envelope already processed in system,skipping. {} ", env.getBasicInfo())
+            );
+        }
+    }
+
+    private boolean canProcess(Optional<Envelope> optionalEnvelope) {
+        return optionalEnvelope.map(e -> Status.CREATED.equals(e.status))
+            .orElse(true);
     }
 
     private void handleBlobWithExistingEnvelope(BlobClient blob, Envelope envelope) {
```

Comment:
probably you won't need this method anymore as there is single line which can moved where this code is used